library/src/main/java/io/getstream/chat/android/client/ChatClientImpl.kt

```diff
@@ -467,6 +468,14 @@ internal class ChatClientImpl(
         notifications.onFirebaseMessage(remoteMessage)
     }
 
+    override fun onMessageHandled(event: NewMessageEvent) {
```

Comment:
Where is this method used?
Is it a method that devs need to use?
Why isn't it used internally when a new message arrive?

Reply:
It's called inside livedata to inform about succesfull cloud message handling (e.g. sync call success). Devs can use it to display a notification about a new message. 

`ChatNotificationHandler` (previously `ChatNotificationConfig`) allows users to intercept cloud message event and handle it on their own. This function allows intercepting the cloud message event, processing it outside (e.g. in livedata sync service), and giving the control back to `ChatNotifications` to display notification as normal. 

This is a bit complex flow, I don't have a good idea how to simplify it. Since all the notifications logic is in the LLC now, but also having sync operation in livedata requires communicating: LLC -> Livedata -> LLC.

Reply:
Another option would be to have additional logic for displaying notifications in livedata, and to remove those functions `onMessageHandled()` and `onMessageHandlingError ()`